### PR TITLE
structure rewrite

### LINKED ISSUE #27: Discussion: file and folder structure
At the moment we use the `minecraft` folder with all fiels inside

i feel it would by more usful to adapt MultiMC's structure
and keep alpacka related files outside of the minecraft folder
i think it will help with implementing features later because we do not have
`feature/feature_name/somefiles` in the minecraft folder where some mod or minecraft itself might mess with it
the gitignore rules will be more compley because we have to ignore the `patches/` folder and `instance.cfg`.. although the latter could be kept in source control if one so desires (not recommended)

### PR DESCRIPTION
**Pull request by NikkyAI**
make every instance use a minecraft folder
closes #27

### CODE REVIEW

File: src/Alpacka.CLI/Commands/CommandInit.cs [hunk 1]
```diff
@@ -16,11 +16,13 @@ public CommandInit()
             Name = "init";
             Description = "Create and initialize a new alpacka pack";
             
+            var argType = Argument("type",
+                "The type of instance can be 'Vanilla', 'Server' or 'MultiMC'");
+            var argName = Argument("name",
+                "The name of the pack");
             var argDirectory = Argument("[directory]",
```
Comment (copygirl) [inline src/Alpacka.CLI/Commands/CommandInit.cs]: Should the directory argument be removed (and we'll just use the name instead?

File: src/Alpacka.CLI/Commands/CommandInit.cs [hunk 2]
```diff
@@ -16,11 +16,13 @@ public CommandInit()
...
                 "The directory to initialize the pack in. Created if necessary.");
             
-            var optName = Option("-n | --name",
-                "Sets the pack name", CommandOptionType.SingleValue);
             var optDescription = Option("-d | --description",
```
Comment (copygirl) [inline src/Alpacka.CLI/Commands/CommandInit.cs]: I.. guess the description and authors should be removed. Whoever uses `init` command will edit the config file manually anyway.

File: src/Alpacka.Lib/Instances/IInstanceHandler.cs [hunk 3]
```diff
@@ -14,6 +14,10 @@ public interface IInstanceHandler
         ///           the specified simplified (safe) instance / folder name. </summary>
         string GetInstancePath(string instanceName);
         
+        /// <summary> Gets the full path of an instance for this type with
+        ///           the specified simplified (safe) instance / folder name. </summary>
+        string GetInstancePath(string instanceName, string basedir);
```
Comment (copygirl) [inline src/Alpacka.Lib/Instances/IInstanceHandler.cs]: Is the old one still used?

File: src/Alpacka.CLI/Commands/CommandInit.cs (hunk 1 above)
Comment (NikkyAI) [inline src/Alpacka.CLI/Commands/CommandInit.cs]: well that would remove some complications

File: src/Alpacka.Lib/Instances/IInstanceHandler.cs (hunk 3 above)
Comment (NikkyAI) [inline src/Alpacka.Lib/Instances/IInstanceHandler.cs]: well all ues of the old one could be changed by passing in the current working directory

**Comment by NikkyAI** [comment]
ping #28 for changes in cli arguments and options

**Pull request merged**